### ISSUE TITLE
Implementation of SimDINOv2 on VIT-Giant(Different loss curves)

### ISSUE THREAD

**Issue by Dojo2024** [RobinWu218/SimDINO#50]
hi @RobinWu218 

i am trying to train simdinov2 using **vit-giant** backbone. i followed the config settings you provided although I'll attatch mine own o. I'm also using pretrained dinov2.

[simdinov2.txt](https://github.com/user-attachments/files/24150118/simdinov2.txt)

 I am seeing very different values in my logs compared to the charts in the readme.
my `dino_mcr_expa_loss` is around **2.5**, but in your logs it is around **0.5**.
also my total `dino_mcr_loss` is very large negative (like **-3** to **-3.5**), but in paper it is closer to **-1.3**.

is this normal because i am using a much bigger model (Giant) and smaller batch size? or did i miss some normalization step? the curves look healthy (going down) but the scale is just very different.

thanks for help.

Here is a snippet of the logs:
<details>
<summary><b>Click here to expand raw logs</b></summary>
I20251214 12:50:15 8353 dinov2 helpers.py:110] Training  [ 4470/18751]  eta: 3:17:39  gap: 8225.72 ms  lr: 0.0004 (0.0002)  wd: 0.0411 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9060 (-3.8146)  dino_mcr_loss: -3.4247 (-3.3749)  dino_mcr_comp_loss: 0.6865 (0.6567)  dino_mcr_global_comp_loss: 0.7725 (0.7389)  dino_mcr_expa_loss: 2.7359 (2.7183)  ibot_comp_loss: -0.4781 (-0.4397)  time: 0.820960  data: 0.000168  max mem: 68905

I20251214 12:50:23 8353 dinov2 helpers.py:110] Training  [ 4480/18751]  eta: 3:17:30  gap: 8259.02 ms  lr: 0.0004 (0.0002)  wd: 0.0411 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9108 (-3.8148)  dino_mcr_loss: -3.4408 (-3.3750)  dino_mcr_comp_loss: 0.6865 (0.6567)  dino_mcr_global_comp_loss: 0.7783 (0.7389)  dino_mcr_expa_loss: 2.7456 (2.7183)  ibot_comp_loss: -0.4781 (-0.4398)  time: 0.824057  data: 0.000167  max mem: 68905

I20251214 12:50:31 8353 dinov2 helpers.py:110] Training  [ 4490/18751]  eta: 3:17:22  gap: 8225.53 ms  lr: 0.0004 (0.0002)  wd: 0.0411 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9007 (-3.8150)  dino_mcr_loss: -3.4201 (-3.3751)  dino_mcr_comp_loss: 0.6631 (0.6567)  dino_mcr_global_comp_loss: 0.7622 (0.7390)  dino_mcr_expa_loss: 2.7456 (2.7184)  ibot_comp_loss: -0.4775 (-0.4399)  time: 0.824039  data: 0.000156  max mem: 68905

I20251214 12:50:40 8353 dinov2 helpers.py:110] Training  [ 4500/18751]  eta: 3:17:13  gap: 8227.12 ms  lr: 0.0004 (0.0002)  wd: 0.0411 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9059 (-3.8152)  dino_mcr_loss: -3.4202 (-3.3753)  dino_mcr_comp_loss: 0.6865 (0.6568)  dino_mcr_global_comp_loss: 0.7686 (0.7390)  dino_mcr_expa_loss: 2.7391 (2.7184)  ibot_comp_loss: -0.4785 (-0.4400)  time: 0.822446  data: 0.000166  max mem: 68905

I20251214 12:50:48 8353 dinov2 helpers.py:110] Training  [ 4510/18751]  eta: 3:17:05  gap: 8197.22 ms  lr: 0.0004 (0.0002)  wd: 0.0411 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9089 (-3.8154)  dino_mcr_loss: -3.4320 (-3.3754)  dino_mcr_comp_loss: 0.6880 (0.6569)  dino_mcr_global_comp_loss: 0.7744 (0.7391)  dino_mcr_expa_loss: 2.7445 (2.7185)  ibot_comp_loss: -0.4802 (-0.4400)  time: 0.821023  data: 0.000173  max mem: 68905

I20251214 12:50:56 8353 dinov2 helpers.py:110] Training  [ 4520/18751]  eta: 3:16:56  gap: 8204.80 ms  lr: 0.0004 (0.0002)  wd: 0.0412 (0.0404)  mom: 0.9003 (0.9001)  last_layer_lr: 0.0004 (0.0002)  current_batch_size: 32.0000 (32.0000)  total_loss: -3.9089 (-3.8156)  dino_mcr_loss: -3.4289 (-3.3755)  dino_mcr_comp_loss: 0.6953 (0.6570)  dino_mcr_global_comp_loss: 0.7710 (0.7392)  dino_mcr_expa_loss: 2.7438 (2.7185)  ibot_comp_loss: -0.4792 (-0.4401)  time: 0.819902  data: 0.000173  max mem: 68905
</details>

<img width="1793" height="774" alt="Image" src="https://github.com/user-attachments/assets/f0a404ec-118a-4614-8bf6-38c04df0e46f" />
